packages/apollo-gateway/src/index.ts

```diff
@@ -108,25 +162,85 @@ export class ApolloGateway implements GraphQLService {
     }
 
     this.initializeQueryPlanStore();
+
+    // this will be overwritten if the config provides one
+    this.updateServiceDefinitions = this.loadServiceDefinitions;
+
+    if (config) {
+      this.updateServiceDefinitions =
+        config.experimental_updateServiceDefinitions ||
+        this.updateServiceDefinitions;
+      // set up experimental observability callbacks
+      this.experimental_didResolveQueryPlan =
+        config.experimental_didResolveQueryPlan;
+      this.experimental_didFailComposition =
+        config.experimental_didFailComposition;
+      this.experimental_didUpdateComposition =
+        config.experimental_didUpdateComposition;
+      this.experimental_pollInterval = config.experimental_pollInterval;
+
+      // ie if they have a pollinterval and a custom loader or a serviceList
+      if (config.experimental_pollInterval) {
+        if (
+          config.experimental_updateServiceDefinitions ||
+          (config as RemoteGatewayConfig).serviceList
+        ) {
+          console.warn(
+            'Polling running services is dangerous and not recommended in production. ' +
+              'Polling should only be used against a registry. ' +
+              'If you are polling running services, use with caution.',
+          );
+        }
+      }
+    }
   }
 
   public async load(options?: { engine?: GraphQLServiceEngineConfig }) {
-    if (options && options.engine) {
-      if (!options.engine.graphVariant)
-        console.warn('No graph variant provided. Defaulting to `current`.');
-      this.engineConfig = options.engine;
-    }
+    const loader = async () => {
+      const previousSchema = this.schema;
+      const previousServiceDefinitions = this.serviceDefinitions;
+      if (options && options.engine) {
+        if (!options.engine.graphVariant)
```

Comment:
wait we have `graphVariant` AND `schemaTag` under engine?

Reply:
I'm not really sure where this came from or if its accurate

Reply:
Worth noting that this is the `engine` option property on the `ApolloGateway` class where it _is_ actually named `graphVariant`, and not the `engine` option property on the `ApolloServer` constructor.

Reply:
🤢